app/src/main/java/org/oppia/android/app/settings/profile/SnackbarController.kt

```diff
@@ -0,0 +1,43 @@
+package org.oppia.android.app.settings.profile
+
+import androidx.annotation.StringRes
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders
+import java.util.LinkedList
+import java.util.Queue
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SnackbarController @Inject constructor(private val dataProviders: DataProviders) {
+
+  private val snackbarRequestQueue: Queue<SnackbarRequest.ShowSnackbar> = LinkedList()
+
+  fun getCurrentSnackbar(): DataProvider<SnackbarRequest> {
+    val currentRequest = snackbarRequestQueue.peek()
+    return dataProviders.createInMemoryDataProvider(SnackbarRequest::class.java) {
```

Comment:
We should use a string ID for parity with other data providers (see any Controller in domain/ that returns a DataProvider).

Reply:
done